src/server.cpp

```diff
@@ -1119,16 +1119,15 @@ PlayerSAO* Server::StageTwoClientInit(u16 peer_id)
 	// Send inventory
 	SendInventory(playersao);
 
-	// Send HP
-	SendPlayerHPOrDie(playersao);
+	// Send HP or death screen
+	if (playersao->isDead())
```

Comment:
SendPlayerHPorDie already check if player is dead and send the death screen + HP. Here you forget to send player HP